core/trino-spi/src/main/java/io/trino/spi/connector/TableProcedureExecutionMode.java

```diff
@@ -20,13 +20,6 @@
 
     public TableProcedureExecutionMode(boolean readsData, boolean supportsFilter)
```

Comment:
it is removed already from here

Reply:
Let's make the constructor private and only have static factories. The boolean arguments are confusing, especially because certain combinations are not valid.

Reply:
You can add Javadoc to the `coordinatorOnly()` factory:

> Table procedure that does not read any table data and only executes on the coordinator.
> Such procedures are useful for custom DDL-type operations.

Reply:
I find the name `distributedWithFilteringAndRepartitioning()` confusing. Would `supportsFiltering=false` also do repartitioning? More generally, how does filtering relate to repartitioning?